sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -327,12 +328,40 @@ internal static void CopyHeaders(HttpHeaders from, HttpHeaders to)
 #if NET6_0_OR_GREATER
         private static string JoinHeaderValues(HeaderStringValues values)
         {
-            return values.Count switch
-            {
-                0 => string.Empty,
-                1 => values.ToString(),
-                _ => string.Join(",", values)
-            };
+            var count = values.Count;
+            // Explicitly handle most common cases
+            switch (count) {
+                case 0:
+                    return string.Empty;
+                case 1:
+                    return values.ToString();
+                case 2:
+                    using (var enumerator = values.GetEnumerator())
+                    {
+                        enumerator.MoveNext();
+                        var value1 = enumerator.Current;
+                        enumerator.MoveNext();
+                        var value2 = enumerator.Current;
+                        return string.Concat(value1, ",", value2);
```

Comment:
Could you shim in string.Create? It seems it would be ideal also for the default

Reply:
Done